src/query/list_file.c

```diff
@@ -96,6 +96,8 @@ typedef struct qfile_list_cache QFILE_LIST_CACHE;
 struct qfile_list_cache
 {
   MHT_TABLE **list_hts;		/* array of memory hash tables for list cache; pool for list_ht of XASL_CACHE_ENTRY */
+  int *free_list;		/* array of freed hash tables */
+  int next;			/* the next freed hash table number */
```

Comment:
It seems queue structure. how about using `std::deque`?
```
#include <deque>

deque<int> free_hash_idx;

// insert
free_hash_idx.push_front (ht_no);

// get value
free_hash_idx.front();

// remove
free.hash_idx.pop_front(); // like stack
free.hash_idx.pop_back(); // like queue
```

Since thread safety is guaranteed by the critical section, it can be safely used.

Reply:
your suggestion is also good, but the current routine is also so simple. I'd like to keep this codes.